text/0082-dynamic-size-region.md

```diff
@@ -61,6 +61,20 @@ balance.
 
 ![dynamic size buckets](../media/dynamic-size-buckets.png)
 
+A new bucket metadata will be added to kvproto:
+```
+message Buckets {
```

Comment:
Should we allocate an ID for each bucket to reduce the size of requests from clients?

Reply:
The ID could consist of term and version of the region and a number.

Reply:
> ...reduce the size of requests from clients?

I don't get it. Client should send the range it wants to scan, so it has to be precisely start key and end key. How can ID reduce the size?

Bucket is supposed to be changed independently from region. For example, client can re-use last overlapping buckets even a region split.

Reply:
It can optimize for queries that need to scan the whole bucket.

Reply:
Because bucket keys are generated by even split, so the range of a query is probably different from the bucket edges.